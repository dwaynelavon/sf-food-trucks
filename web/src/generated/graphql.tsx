import gql from 'graphql-tag';
export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type FoodTruck = {
  __typename?: 'FoodTruck';
  objectId?: Maybe<Scalars['String']>;
  applicant?: Maybe<Scalars['String']>;
  facilityType?: Maybe<Scalars['String']>;
  cnn?: Maybe<Scalars['String']>;
  locationDescription?: Maybe<Scalars['String']>;
  address?: Maybe<Scalars['String']>;
  blocklot?: Maybe<Scalars['String']>;
  block?: Maybe<Scalars['String']>;
  lot?: Maybe<Scalars['String']>;
  permit?: Maybe<Scalars['String']>;
  status?: Maybe<Scalars['String']>;
  foodItems?: Maybe<Scalars['String']>;
  x?: Maybe<Scalars['String']>;
  y?: Maybe<Scalars['String']>;
  latitude?: Maybe<Scalars['String']>;
  longitude?: Maybe<Scalars['String']>;
  schedule?: Maybe<Scalars['String']>;
  received?: Maybe<Scalars['String']>;
  priorPermit?: Maybe<Scalars['String']>;
  expirationDate?: Maybe<Scalars['String']>;
  location?: Maybe<Location>;
  sourceLat?: Maybe<Scalars['Float']>;
  sourceLng?: Maybe<Scalars['Float']>;
  sourceDistance?: Maybe<Scalars['Float']>;
};

export type Query = {
  __typename?: 'Query';
  foodTrucksNearby: Array<FoodTruck>;
};


export type QueryFoodTrucksNearbyArgs = {
  lat: Scalars['Float'];
  lng: Scalars['Float'];
};

export type Location = {
  __typename?: 'Location';
  type?: Maybe<Scalars['String']>;
  coordinates?: Maybe<Array<Maybe<Scalars['Float']>>>;
};


