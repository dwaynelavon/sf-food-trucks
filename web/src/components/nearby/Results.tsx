import React from "react";
import { FoodTruck } from "../../generated/graphql";
import ResultCard from "./ResultCard";

interface IProps {
    foodTrucks: FoodTruck[];
    loading: boolean;
}

const renderFoodTruckResults = (res: FoodTruck[]) =>
    res.map(({ applicant, locationDescription, foodItems, sourceDistance }) => (
        <ResultCard
            title={applicant!}
            description={foodItems!}
            distance={sourceDistance!}
            location={locationDescription!}
        />
    ));

const Results = ({ foodTrucks, loading }: IProps) => {
    return (
        <div className="results">
            {loading ? (
                <progress className="progress is-small is-primary" max="100">
                    15%
                </progress>
            ) : (
                renderFoodTruckResults(foodTrucks)
            )}
        </div>
    );
};

export default Results;
