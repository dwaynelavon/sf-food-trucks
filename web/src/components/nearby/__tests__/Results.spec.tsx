import { shallow } from "enzyme";
import React from "react";
import Results from "../Results";
import { FoodTruck } from "../../../generated/graphql";

describe("Results", () => {
    it("should render with default props", () => {
        const foodTruck: FoodTruck = {
            locationDescription: "truck is here",
            objectId: "abc123",
            applicant: "My Truck",
            sourceDistance: 0.45,
            foodItems: "tacos sushi and pizza",
        };
        const props = {
            loading: false,
            foodTrucks: [foodTruck],
        };
        const wrapper = shallow(<Results {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
