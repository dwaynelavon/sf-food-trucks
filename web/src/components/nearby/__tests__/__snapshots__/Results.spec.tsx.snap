// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render with default props 1`] = `ShallowWrapper {}`;
